src/main.c

```diff
@@ -1222,9 +1222,56 @@ static int on_config_max_connections(h2o_configurator_command_t *cmd, h2o_config
 
 static int on_config_num_threads(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
 {
-    if (h2o_configurator_scanf(cmd, node, "%zu", &conf.num_threads) != 0)
+    size_t num_threads = 0;
+    free(conf.thread_map.entries);
+    memset(&conf.thread_map, 0, sizeof(conf.thread_map));
```

Comment:
While these two lines work, simply setting `conf.thread_map.size` to zero would be sufficient.